subprojects/plugin-use/src/main/java/org/gradle/plugin/management/internal/DefaultPluginManagementPluginResolver.java

```diff
@@ -0,0 +1,37 @@
+package org.gradle.plugin.management.internal;
+
+import org.gradle.plugin.use.PluginId;
+import org.gradle.plugin.use.internal.InternalPluginRequest;
+import org.gradle.plugin.use.internal.InvalidPluginRequestException;
+import org.gradle.plugin.use.resolve.internal.PluginResolution;
+import org.gradle.plugin.use.resolve.internal.PluginResolutionResult;
+import org.gradle.plugin.use.resolve.internal.PluginResolveContext;
+
+public class DefaultPluginManagementPluginResolver implements PluginManagementPluginResolver {
```

Comment:
Plugin management is not a new resolver. It is always executed on the plugin requests that were made by the user, potentially replacing them with different/enriched requests. Then these new requests are passed on to the existing resolvers. The existing resolvers can then make new decisions based on the richer request data. E.g. the Maven/Ivy resolver can look if the an explicit dependencyNotation was set and use that one instead of using their default logic.